src/pages/subreddit/Subreddit.js

```diff
@@ -1,40 +1,48 @@
 import React, { useEffect, useState } from 'react';
 import './Subreddit.css';
 import axios from 'axios';
-import {Link, useHistory} from "react-router-dom";
+import {useParams} from "react-router-dom";
 
 
 function Subreddit() {
-    const history = useHistory();
-    const [subreddit, setSubreddit] = useState('');
+    const [subreddit, setSubreddit] = useState({});
+    const { id } = useParams();
+    // zou ik ook meerdere useParams in kunnen stellen of altijd maar 1?
 
     useEffect(() => {
         async function fetchCategories() {
             try {
-                const response = await axios.get('https://www.reddit.com/r/{id}/about.json');
-                console.log(response.data.data.children);
-                setSubreddit(response.data.data.children);
+                const response = await axios.get(`https://www.reddit.com/r/${id}/about.json`);
+                // Is deze link een speciale link waarbij je de data kunt ophalen via het ID of kun je
+                // dat bij een willekeurige api link ook doen met het toevoegen van de {$id} ?
+                console.log(response);
+                setSubreddit(response.data.data);
 
             } catch(e) {
                 console.error(e);
             }
         }
         fetchCategories();
-    }, []);
+    },);
 
     return (
-        <section className="subreddit">
-            <div>
-                {subreddit((reddits) => {
-                    return (
-                        <>
-                            <h3 className="title" key={reddits.data}>{reddits.data.title}</h3>
-                        </>
-                    )
-                })}
+        <>
+        <main>
+            <section className="reddit-card">
+                               {Object.keys(subreddit).length> 0 && (
```

Comment:
Je wilt hier eigenlijk kunnen weten dat het `subreddit` al is opgehaald met axios. Je kan beter onderscheid maken als je de default waarde `undefined` maakt. Dan kun je dit een stuk versimpelen
```suggestion
                               {subreddit && (
```